### PR TITLE
Add man page and tutorial

### PR DESCRIPTION
**Pull request by meme**
https://git.sr.ht/~sircmpwn/scdoc is used for document generation, and
the format of these pages is inspired by the aerc documentation.

The tutorial is inspired by the cmus-tutorial provided by the C* Music
Player

@merces, I'd appreciate a review on this when you have a moment :smile:.

You can generate and view the man pages with:
```
cd man
scdoc < hexxed-tutorial.7.scd > hexxed-tutorial.7
groff -Tascii -man hexxed-tutorial.7 | less -R
```

I will update the build system to support building and installing these pages afterwards.

### CODE REVIEW

**Comment by merces** [comment]
Wow! Just read over it and enjoyed it a lot! Thanks!

File: man/hexxed-tutorial.7.scd [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+are displayed on the screen. See *isprint(3)* for more information. Hit *Enter*
+to cycle back to the *Hex* view
+
+# Step 5: Editing
+
+Hexxed is fundamentally a hex editor, so it is straightforward to perform
+edit operations. Hit *F3* to enter edit mode, and the cursor will change to select
+a single *nibble* instead of a *pair*. To overwrite the data under the cursor,
+enter a character in the range *0-9* or *a-f*. The change will be reflected in the
+buffer. Hit *Escape* to exit.
```
Comment (merces) [inline man/hexxed-tutorial.7.scd:115]: Do you think it should be clearer that changes will be written to the file? I don't know, I feel like hitting **Escape** suggests "operation cancelled" in a certain way but maybe it's just me. 🙂

Comment (meme) [reply man/hexxed-tutorial.7.scd:115]: You make a good point, I should rewrite the edit mode to undo edits if Escape is hit, and save edits if F3 is hit again (and exit edit mode). Does this make more sense?

Comment (merces) [reply man/hexxed-tutorial.7.scd:115]: Yes, I think so. I tried to look for a pattern, but apparently there is none. =/

[HIEW ](http://hiew.ru/)uses `F3` to enter editing mode:

hiew_before

But then hitting `F3` again will undo the edits. `Escape` also undo them. Only `F9` saves the changes.

hiew_edit_mode

[HT Editor](http://hte.sourceforge.net/) uses `F4` for entering editing mode:

hte_before

 `Escape` undo the changes while `F2` save them:

hte_editing_mode

In general, I'd suggest to try your best to make shortcut keys compatible to existent software but in that case is a hard decision. HIEW is famous in malware analysis field, but it's paid. HT is open source, but it seems to be abandoned. hexxed is our new hope. :)

**Review by merces — APPROVED** [review]
Extremely useful. As a new hexxed user, I appreciate that a lot!

**Review by merces — APPROVED** [review]

File: man/hexxed-tutorial.7.scd (hunk 1 above)
Comment (meme) [reply man/hexxed-tutorial.7.scd:115]: Will start tracking this here: https://github.com/meme/hexxed/issues/8. Thanks for providing a comparison with other editors.

**Comment by meme** [comment]
Appreciate the review, thanks!

**Pull request merged**